src/SFA.DAS.PR.Jobs/Functions/Notifications/SendNotificationsFunction.cs

```diff
@@ -0,0 +1,103 @@
+        ITokenService tokenService
+    )
+    {
+        _logger = logger;
+        _functionEndpoint = functionEndpoint;
+        _providerRelationshipsDataContext = providerRelationshipsDataContext;
+        _notificationRepository = notificationRepository;
+        _tokenService = tokenService;
+
+        _notificationsConfiguration = new NotificationsConfiguration();
+        configuration.GetSection("ApplicationConfiguration:Notifications").Bind(_notificationsConfiguration);
+    }
+
+    [Function(nameof(SendNotificationsFunction))]
+    public async Task<int> Run([TimerTrigger("%SendNotificationsFunctionSchedule%", RunOnStartup = true)] TimerInfo timer, FunctionContext executionContext, CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("{FunctionName} has been triggered.", nameof(SendNotificationsFunction));
+
+        List<Notification> notifications = await _notificationRepository.GetPendingNotifications(
+            _notificationsConfiguration.BatchSize, 
+            NotificationType.Provider, 
+            cancellationToken
+        );
+
+        if(notifications.Count == 0)
+        {
+            return 0;
+        }
+
+        int processedCount = 0;
+
+        foreach (Notification notification in notifications)
+        {
+            processedCount += await ProcessNotification(notification, executionContext, cancellationToken);
+        }
+
+        await _providerRelationshipsDataContext.SaveChangesAsync(cancellationToken);
+
+        _logger.LogInformation("{FunctionName} - Processed {ProcessedCount} notifications.", nameof(SendNotificationsFunction), processedCount);
+
+        return processedCount;
+    }
+
+    private async Task<int> ProcessNotification(Notification notification, FunctionContext executionContext, CancellationToken cancellationToken)
+    {
+        try
+        {
+            SendEmailCommand sendEmailCommand = await CreateSendEmailCommand(notification, cancellationToken);
+            await _functionEndpoint.Send(sendEmailCommand, executionContext, cancellationToken);
+            notification.SentTime = DateTime.UtcNow;
+
+            return 1;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Exception sending out notification with id: {notificationId}", notification.Id);
+
+            return 0;
+        }
+    }
+
+    private async Task<SendEmailCommand> CreateSendEmailCommand(Notification notification, CancellationToken cancellationToken)
+    {
+        TemplateConfiguration? templateConfiguration = _notificationsConfiguration.NotificationTemplates.Find(a => a.TemplateName == notification.TemplateName);
+
+        if(templateConfiguration == null || string.IsNullOrWhiteSpace(templateConfiguration.TemplateId))
+        {
+            throw new ArgumentNullException($"Template {notification.TemplateName} does not exist in the configuration for notification {notification.Id}.");
+        }
```

Comment:
This should never happen and hence we should not code around it. If it happens, then it will naturally fail and we should let that exception bubble up.